packages/react/menu/src/Menu.tsx

```diff
@@ -956,6 +1012,30 @@ function getNextMatch(values: string[], search: string, currentMatch?: string) {
   return nextMatch !== currentMatch ? nextMatch : undefined;
 }
 
+type Point = { x: number; y: number };
+type Triangle = [Point, Point, Point];
+
+function getTriangleArea(a: Point, b: Point, c: Point) {
+  return Math.abs((a.x * (b.y - c.y) + b.x * (c.y - a.y) + c.x * (a.y - b.y)) / 2);
+}
+
+function isPointInTriangle(point: Point, a: Point, b: Point, c: Point) {
+  const A = getTriangleArea(a, b, c);
+  const A1 = getTriangleArea(point, b, c);
+  const A2 = getTriangleArea(a, point, c);
+  const A3 = getTriangleArea(a, b, point);
+  return A === A1 + A2 + A3;
+}
+
+function isMouseMovingToSubmenu(event: React.MouseEvent, triangle: Triangle | null) {
```

Comment:
In my mind the name `isMouseMovingToSubmenu` could be `true` under broader terms like just leaving the sub trigger and not hovering another sibling item so I was going to suggest renaming it to `isPointerWithinGraceArea` and then I saw we already have that.

So now, I'm actually wondering why we need this abstraction at all... can't we just call `isPointInTriangle` (maybe rename to `isPointInGraceArea` for consistency) instead?

```jsx
const pointerPos = { x: event.clientX, y: event.clientY };
if (isPointInGraceArea(pointerPos, ...pointerGraceAreaRef.current)) {}
```

Reply:
We can, but we put it in a util as it's used 3 times cause otherwise each time you'd need to construct a point, handle the null case, and spread the ref.

Reply:
I personally don't feel that re-constructing a point and spreading a ref warrant this entire abstraction. In fact, we wouldn't even need to spread and the `null` case would just be this no?

```jsx
function isPointInGraceArea(pointerPoint, graceArea: Triangle | null) {
  if (!graceArea) return false;
  const [a, b, c] = graceArea;
  const A = getTriangleArea(a, b, c);
  const A1 = getTriangleArea(pointerPoint, b, c);
  const A2 = getTriangleArea(a, pointerPoint, c);
  const A3 = getTriangleArea(a, b, pointerPoint);
  return A === A1 + A2 + A3;
}
```

Reply:
I'd still keep both utils to make things clean rather than flattening, but otherwise I'm ok with it.

```tsx
function isPointerInGraceArea(pointerPos: Point, graceArea: Triangle | null) {
  if (!graceArea) return false;
  return isPointInTriangle(pointerPos, ...graceArea);
}
```